General improvement to CLI reference for Teleport daemon and more (#15974)
gravitational/teleport: docs/pages/reference/cli.mdx

---
title: Teleport CLI Reference
description: Detailed guide and reference documentation for Teleport's command line interface (CLI) tools.
---

Teleport is made up of four CLI tools.

> - [teleport](#teleport): A CLI tool that supports the Teleport access plane by starting and configuring various Teleport services.

Reviewer: I don't think we need to repeat "CLI tool" in each list item, since we already say that each list item describes a CLI tool.

- [tsh](#tsh): A client CLI tool that allows end users to authenicate to Teleport and access resources in a cluster.
- [tctl](#tctl): An administrative CLI tool that can configure the Teleport Auth Service.
- [tbot](#tbot): A CLI tool that supports Machine ID, which provides short lived credentials to service accounts (e.g, a CI/CD server).

(!docs/pages/includes/permission-warning.mdx!)

(!docs/pages/includes/backup-warning.mdx!)

## teleport

The CLI tool that supports the Teleport access plane is called `teleport`, and allows Teleport services to be managed
over the command line:

> - [Auth](../architecture/authentication.mdx)

Reviewer: I know it would repeat "Service", but the full name of each service is the "Auth Service", "Proxy Service", etc. Otherwise, I'm not sure it's clear what a user should expect each link in this list to point to.

Reviewer: @rsyracuse Just checking to see if you wanted to incorporate this one

- [Node/SSH](../architecture/nodes.mdx)
- [Proxy](../architecture/proxy.mdx)
- [App](../application-access/introduction.mdx)
- [Database](../database-access/introduction.mdx)
- [Windows Desktop](../desktop-access/introduction.mdx)
- [Kubernetes](../kubernetes-access/introduction.mdx)

> - [Generate a systemd unit file for a systemd managed service daemon](../management/admin/daemon.mdx).

Reviewer: All of the items in this list are links to documentation about a Teleport service except for this link and the one below. Why are these links here?

- [Generate a sample configuration YAML file for an SSH Node](../architecture/nodes.mdx).

The primary commands for the `teleport` CLI are as follows:

| Command | Description |
| - | - |
| `teleport help` | Outputs guidance for using Teleport commands. |
| `teleport start` | Starts the `teleport` process in the foregound using the current shell session, including any services configured by the [configuration YAML file](config.mdx). |
| `teleport status` | Prints the status of the current active Teleport SSH session. |
| `teleport configure` | Generates and writes a [configuration YAML file](config.mdx) for the Teleport service. This file should be customized in production to suit the needs of your environment, and the default output should only be used when testing. |
| `teleport version` | Prints the current release version of the Teleport binary installed on your system. |

> | `teleport app start` | Starts the application proxy service. |

Reviewer: ```suggestion
| `teleport app start` | Starts the Teleport Application Service. |
```

| `teleport db configure create` | Generates a configuration YAML file for the Database Service. This file should be customized in production to suit the needs of your environment, and the default output should only be used when testing. |
| `teleport db configure bootstrap` | Used to bootstrap a configuration to the Teleport Database Service by reading a provided configuration. |
| `teleport db configure aws print-iam` | Generates and outputs current IAM policies for a Teleport-managed database. |
| `teleport db configure aws create-iam` | Generates, creates, and attaches desired IAM policies to a Teleport-managed database. |
| `teleport install systemd` | Creates a systemd unit file, used to configure and install a `teleport` service daemon. |
| `teleport node configure` | Generates a configuration YAML file for a Teleport Node accessed via SSH. This file should be customized in production to suit the needs of your environment, and the default output should only be used when testing. |

<Notice type="tip">
For more information on subcommands when working with the `teleport` cli, use the `--help` option or `teleport <subcommand> --help`.
</Notice>

### teleport start

> The `teleport start` command includes a large number of optional configuration flags which change the behavior of the default command to support a wider spectrum of Teleport services.

Reviewer: ```suggestion
The `teleport start` command includes a large number of optional configuration flags.
```
I think we can make this sentence much more concise. Do we need more than this part of it? For me, the crux of the paragraph is that `teleport start` is unwieldy and using a configuration file is preferred.

---
title: Teleport CLI Reference
description: Detailed guide and reference documentation for Teleport's command line interface (CLI) tools.
---

Teleport is made up of four CLI tools.

- [teleport](#teleport): Supports the Teleport access plane by starting and configuring various Teleport services.

> - [tsh](#tsh): Allows end users to authenicate to Teleport and access resources in a cluster.

Reviewer: ```suggestion
- [tsh](#tsh): Allows end users to authenticate to Teleport and access resources in a cluster.
```

h1: Command Line (CLI) Reference
---

Teleport is made up of four CLI tools.

- [teleport](#teleport): A process and CLI tool that supports the Teleport access plane.
- [tsh](#tsh): A CLI tool that allows end users to interact with Teleport sessions and list cluster information.
- [tctl](#tctl): An administrative CLI tool that can configure the Teleport Auth Service.

> - [tbot](#tbot): A CLI tool that supports Machine ID on Teleport, used to apply short lvied certificates to service accounts (e.g, a CI/CD server).

Reviewer 2: ```suggestion
- [tbot](#tbot): A CLI tool that supports Machine ID on Teleport, used to apply short lived certificates to service accounts (e.g, a CI/CD server).
```

h1: Command Line (CLI) Reference
---

Teleport is made up of four CLI tools.

- [teleport](#teleport): A process and CLI tool that supports the Teleport access plane.
- [tsh](#tsh): A CLI tool that allows end users to interact with Teleport sessions and list cluster information.
- [tctl](#tctl): An administrative CLI tool that can configure the Teleport Auth Service.
- [tbot](#tbot): A CLI tool that supports Machine ID on Teleport, used to apply short lived certificates to service accounts (e.g, a CI/CD server).

(!docs/pages/includes/permission-warning.mdx!)

(!docs/pages/includes/backup-warning.mdx!)

## teleport

> The CLI tool that supports the Teleport access plane is called `teleport`, and allows Teleport services to be created, configured, and otherwise interacted with over the command line.

Reviewer: - Same comment as above re: "supports the Teleport access plane"
- What does it mean for a Teleport service to be "created"? Should we say instead that it runs Teleport services, and list the services that it runs?

The `teleport` process' CLI capability is generally the first step in setting up the Teleport binary following package manager or source installation on Linux.

The primary commands for the teleport CLI are as follows:

| Command | Description |
| - | - |
| `teleport help` | Outputs guidance for using commands on Teleport. |
| `teleport start` | Starts the `teleport` service using the current shell session. This command will not start the `teleport` service as a daemon. |
| `teleport status` | Prints the status of the current active Teleport SSH session. |
| `teleport configure` | Generates and writes a [configuration YAML file](config.mdx) for the Teleport service. This file should be customized in production to suit the needs of your environment, and the default output should only be used when testing. |
| `teleport version` | Prints the current release version of the Teleport binary installed on your system. |
| `teleport app start` | Starts the application proxy service. |

> | `teleport db start` | Starts the database proxy service. |

Reviewer: ```suggestion
| `teleport db start` | Starts the Teleport Database Service. |
```

| `teleport db configure create` | Generates a database service configuration. |

> | `teleport db configure bootstrap` | Used to bootstrap a configuration to the Teleport database agent by reading a provided configuration. |

Reviewer: ```suggestion
| `teleport db configure bootstrap` | Used to bootstrap a configuration to the Teleport Database Service by reading a provided configuration. |
```

> | `teleport db configure aws print-iam` | Generates and outputs current IAM policies for a teleport managed database. |

Reviewer: ```suggestion
| `teleport db configure aws print-iam` | Generates and outputs current IAM policies for a Teleport-managed database. |
```

> | `teleport db configure aws create-iam` | Generates, creates, and attaches desired IAM policies to a teleport managed database. |

Reviewer: ```suggestion
| `teleport db configure aws create-iam` | Generates, creates, and attaches the desired IAM policies to a Teleport-managed database. |
```

| `teleport install systemd` | Creates a systemd unit file, used to configure and install a `teleport` service daemon. |

> | `teleport node configure` | Generates a configuration file for a Teleport node accessed via SSH. |

Reviewer: ```suggestion
| `teleport node configure` | Generates a configuration file for a Teleport Node accessed via SSH. |
```
To distinguish this from the general term "node" (https://goteleport.com/docs/architecture/overview/#definitions).

<Notice type="tip">

> For more information on subcommands when working with the `teleport` cli, use the `--help` option, or `teleport <subcommand> --help`.

Reviewer: ```suggestion
For more information on subcommands when working with the `teleport` CLI, use the `--help` option or `teleport <subcommand> --help`.
```

> Teleport is made up of three CLI tools.

Reviewer 2: I know this in the original doc, but why don't we mention `tbot`?

> - [teleport](#teleport): A CLI tool that supports the Teleport access plane.

Reviewer 3: It's a little odd to hear `teleport` described as a CLI tool, but I suppose it's not entirely wrong to say so. `daemon` was definitely more misleading as Teleport runs as a foreground process. So perhaps this will do for now.

Reviewer 4: I struggled with the wording on this as well, but felt that it was best to lean into CLI ability given the context of the doc. Maybe it'd be better to say "A CLI tool and foreground process..." just to be more accurate?

- [tsh](#tsh): A CLI tool that allows end users to interact with Teleport sessions and list cluster information.

> - [tctl](#tctl): An administrative tool that can configure the Teleport Auth Service.

Reviewer 3: If the other two are described as a CLI tool, `tctl` certainly should be as well.

| `teleport version` | Prints the current release version of the Teleport binary installed on your system. |
| `teleport app start` | Starts the application proxy service. |
| `teleport db start` | Starts the database proxy service. |
| `teleport db configure create` | Generates a database service configuration. |
| `teleport db configure bootstrap` | Used to bootstrap a configuration to the Teleport database agent by reading a provided configuration. |
| `teleport db configure aws print-iam` | Generates and outputs current IAM policies for a teleport managed database. |
| `teleport db configure aws create-iam` | Generates, creates, and attaches desired IAM policies to a teleport managed database. |
| `teleport install systemd` | Creates a systemd unit file, used to configure and install a `teleport` service daemon. |
| `teleport node configure` | Generates a configuration file for a Teleport node accessed via SSH. |

<Notice type="tip">
For more information on subcommands when working with the `teleport` cli, use the `--help` option, or `teleport <subcommand> --help`.
</Notice>

### teleport start

> The `teleport start` command includes a large number of optional configuration flags which change the behavior of the default command to support a wider spectrum of teleport products and services.

Reviewer 3: This might be a good place to guide people towards *not* using those configuration flags and using the YAML config files unless they are experimenting with Teleport features.

### teleport start

The `teleport start` command includes a large number of optional configuration flags.

<Admonition title="Configuration Flags" type="tip">

> While configuration flags for `teleport start` can be used to set parameters for Teleport's configuration,

Reviewer: I think we should either move the "The `teleport start` command..." sentence into this Admonition or turn this Admonition into body text, since I don't think the first sentence makes sense outside of the recommendation to use a configuration file. Otherwise, it just sounds like we're passing judgment on the number of optional configuration flags.

| `--app-uri` | none | none | Internal address of the application to proxy |
| `--app-public-addr` | none | none | Public address fo the application to proxy |

#### teleport start --roles

The `--roles` flag when used with `teleport --start` instructs Teleport on which specific Teleport services to start. Below is a more cohesive table of roles and their associated services that `teleport start` supports:

| Service | Role Name | Description |
| - | - | - |
| [Node](../architecture/nodes.mdx) | `node` | Runs a daemon on a host that allows SSH connections from authenticated clients. |

> | [Auth](../architecture/authentication.mdx) | `auth` | Authenticates hosts and users who want access to Teleport-managed resources or information about a cluster. |

Reviewer 2: ```suggestion
| [Auth](../architecture/authentication.mdx) | `auth` | Authenticates and authorizes hosts and users who want access to Teleport-managed resources or information about a cluster. |
```

| [Proxy](../architecture/proxy.mdx) | `proxy` | The gateway that clients use to connect to the Auth Service or resources managed by Teleport. |

> | [App](../application-access/introduction.mdx) | `app` | Runs a daemon on a host that provides access to applications using an SSH reverse tunnel. |

Reviewer 2: ```suggestion
| [App](../application-access/introduction.mdx) | `app` | Provides access to applications. |
```

Remove "daemon", and I'm not sure the reverse tunnel is relevant info - kind of an implementation detail, although if you want to emphasize the consequence of that detail (Teleport only requires the proxy service to be externally reachable thanks to reverse tunnels) then that would be useful info.

> | [Kube](../kubernetes-access/introduction.mdx) | `kube` | The Teleport daemon will run the Kubernetes Service. |

Reviewer 2: ```suggestion
| [Kube](../kubernetes-access/introduction.mdx) | `kube` | Provides access to Kuberenetes clusters. |
```

"daemon" here

> | [DB](../database-access/reference.mdx) | `db` | The Teleport daemon will run the Database Service. |

Reviewer 2: ```suggestion
| [DB](../database-access/reference.mdx) | `db` | Provides access to databases. |
```

"daemon" here

> | [Trusted Cluster](../management/admin/trustedclusters.mdx) | `trusted_cluster` | The Teleport daemon will support a leaf cluster used to connect to another Teleport cluster. |

Reviewer 2: Is `trusted_cluster` a role now? I don't see it in the master branch lib/config/configuration.go:validateRoles func.

> | [Windows Desktop](../desktop-access/getting-started.mdx) | `windowsdesktop` | The Teleport daemon will run the Windows Desktop Service. |

Reviewer 2: ```suggestion
| [Windows Desktop](../desktop-access/getting-started.mdx) | `windowsdesktop` | Provides Windows desktop access. |
```

"daemon" here

| `--init-dir`        | If using a config file and multiple destinations are configured, controls which destination dir to configure.      |
| `--clean`           | If set, remove unexpected files and directories from the destination.                                              |

#### Examples

**Example using file permissions.**

The following command highlights how to set permissions with tbot through Linux groups, using the user and group `jenkins:jenkins`.
If running `tbot` as the Linux user `root`, use the following invocation of
`tbot init` to initialize the short-lived certificate directory
`/opt/machine-id` with owner `jenkins:jenkins`.

```code
$ tbot init \
    --destination-dir=/opt/machine-id \
    --owner=jenkins:jenkins
```

**Example using POSIX ACLs.**

If running `tbot` as the Linux user `teleport`, use the following invocation of
`tbot init` to initialize the short-lived certificate directory
`/opt/machine-id` with owner `teleport:teleport` but allow `jenkins` to read
from `/opt/machine-id`.

```code
$ tbot init \
    --destination-dir=/opt/machine-id \
    --bot-user=teleport \
    --reader-user=jenkins
```

### tbot db

> Allows connections to databases using native clients and queries database information. This is best used for testing and validation purposes;

Reviewer 2: ```suggestion
Connects to databases using native clients and queries database information. This is best used for testing and validation purposes;
```

nit: I would rephrase like this, since `tbot db` isn't allowing connections, it's just connecting

| `--destination-dir` | Path to the Machine ID destination dir that should be used for authentication. Required.                 |
| `--proxy`           | The `host:port` of the Teleport Proxy Service to use to access resources. Required.                      |
| `--cluster`         | The name of the cluster on which resources should be accessed. Extracted from the bot identity if unset. |

All other flags and arguments are passed directly to `tsh db ...`, along
with authentication parameters to use the Machine ID identity to skip `tsh`'s
login steps.

Note that certain CLI parameters, for example `--help`, may be captured by
`tbot` even if intended to be passed to the wrapped `tsh`. A `--` argument can
be used to ensure all following arguments are passed to `tsh` and ignored by
`tbot`.

Additionally, be aware of the following limitations of `tbot db`:
 - `tbot db connect` requires a `tbot db login` for certain database types,
   like MySQL, so that additional connection parameters can be written to a
   local configuration file.
 - `tbot db env` is not fully supported.

### tbot proxy

> Allows for access to teleport resources via a proxy using various protocols. Requires running a local ALPN/SNI proxy to access the resources if using TLS.

Reviewer: ```suggestion
Allows for access to Teleport resources via a proxy using various protocols. Requires running a local ALPN/SNI proxy to access the resources if using TLS.
```

Reviewer: It's difficult for me to tell what this command does from this paragraph. 

- What does "via a proxy" mean? What kind of proxy is this? How would a user know whether a given proxy is relevant to this command?

- How do you know whether you have the required local ALPN/SNI proxy running? Also, `tsh proxy db` starts the local ALPN/SNI proxy for you—does this command do that as well?

The `tbot proxy` command acts as a wrapper for `tsh proxy` to provide local proxy functionality for various protocols.

Note that tsh must be installed to make use of this command.

#### Flags

| Flag                | Description                                                                                              |
|---------------------|----------------------------------------------------------------------------------------------------------|
| `-d/--debug`        | Enable verbose logging to stderr.                                                                        |

> | `-c/--config`       | Path to a configuration file.                                                                            |

Reviewer: Is this required? If not, what's the default?

Reviewer: I'm still not sure what the default value for this is. Also, is this a `tbot` configuration profile or a `tsh` configuration file?

# with key `env` equal to `staging` and key `os` equal to `mac`.
$ tsh ls --query='labels["env"] == "staging" && equals(labels["os"], "mac")'
```

## tbot

`tbot` is a CLI tool used with **Machine ID** that programatically issues and renews
short-lived certificates to any service account (e.g, a CI/CD server).

The primary commands for `tbot` are as follows:

| Command | Description |
| - | - |

> | `tbot help` | Outputs guidance for using commands with `tbot` |

Reviewer: ```suggestion
| `tbot help` | Outputs guidance for using commands with `tbot`. |
```
So punctuation is consistent in this column

| `tbot version` | Outputs the current version of the `tbot` binary. |
| `tbot configure` | Outputs a basic Machine ID client configuration file to be adjusted as needed. |
| `tbot start` | Starts the Machine ID client `tbot`, fetching and writing certificates to disk at a set interval. |
| `tbot init` | Initialize a certificate destination directory for writes from a separate bot user, configuring either file or POSIX ACL permissions. |
| `tbot db` | Connects to databases using native clients and queries database information. Functions as a wrapper for `tsh`, and requires `tsh` installation. |
| `tbot proxy` | Allows for access to Teleport resources on a cluster using TLS routing. Functions as a wrapper for `tsh`, and requires `tsh` installation. |

### tbot start

> Starts the Machine ID client tbot, fetching and writing certificates to disk at a set interval.

Reviewer: ```suggestion
Starts the Machine ID client `tbot`, fetching and writing certificates to disk at a set interval.
```

| `--token`            | A bot join token, if attempting to onboard a new bot; used on first connect. Can also be an absolute path to a file containing the token. |
| `--ca-pin`           | CA pin to validate the Teleport Auth Server; used on first connect.                            |
| `--data-dir`         | Directory to store internal bot data. In production environments access to this directory should be limited only to an isolated linux user as an owner with `0600` permissions. |
| `--destination-dir`  | Directory to write short-lived machine certificates.                                           |
| `--certificate-ttl`  | TTL of short-lived machine certificates.                                                       |
| `--renewal-interval` | Interval at which short-lived certificates are renewed; must be less than the certificate TTL. |
| `--join-method`      | Method to use to join the cluster. Can be `token` or `iam`.                                    |
| `--oneshot`          | If set, quit after the first renewal.                                                          |

#### Examples

> <ScopedBlock scope={["cloud"]}>

Reviewer: I think we should use a `Tabs` component instead in case a user has forgotten to select a scope (which is most of the time).

```code
$ tbot start \
   --data-dir=/var/lib/teleport/bot \
   --destination-dir=/opt/machine-id \
   --token=00000000000000000000000000000000 \
   --join-method=token \
   --ca-pin=sha256:1111111111111111111111111111111111111111111111111111111111111111 \
   --auth-server=example.teleport.sh:443
```

</ScopedBlock>

<ScopedBlock scope={["oss","enterprise"]}>

```code
$ tbot start \
   --data-dir=/var/lib/teleport/bot \
   --destination-dir=/opt/machine-id \
   --token=00000000000000000000000000000000 \
   --join-method=token \
   --ca-pin=sha256:1111111111111111111111111111111111111111111111111111111111111111 \
   --auth-server=auth.example.com:3025
```

</ScopedBlock>

### tbot init

> Allows certificates to be written to disks other than a Machine ID client, configuring either file or POSIX ACL permissions.

Reviewer: What does `tbot init` do? This sentence just describes its effects.

| `-d/--debug`        | Enable verbose logging to stderr.                                                                                  |
| `-c/--config`       | Path to a configuration file.                                                                                      |
| `--destination-dir` | Directory to write short-lived machine certificates to.                                                            |
| `--owner`           | Defines the Linux `user:group` owner of `--destination-dir`. Defaults to the Linux user running `tbot` if unspecified. |
| `--bot-user`        | Enables POSIX ACLs and defines the Linux user that can read/write short-lived certificates to `--destination-dir`. |
| `--reader-user`     | Enables POSIX ACLs and defines the Linux user that will read short-lived certificates from `--destination-dir`.    |
| `--init-dir`        | If using a config file and multiple destinations are configured, controls which destination dir to configure.      |
| `--clean`           | If set, remove unexpected files and directories from the destination.                                              |

#### Examples

**Example using file permissions.**

> The following command highlights how to set permissions with tbot through Linux groups, using the user and group `jenkins:jenkins`.

Reviewer: ```suggestion
The following command highlights how to set permissions with `tbot` through Linux groups, using the user and group `jenkins:jenkins`.
```

| `--proxy`           | The `host:port` of the Teleport Proxy Service to use to access resources. Required.                      |
| `--cluster`         | The name of the cluster on which resources should be accessed. Extracted from the bot identity if unset. |

All other flags and arguments are passed directly to `tsh db ...`, along
with authentication parameters to use the Machine ID identity to skip `tsh`'s
login steps.

Note that certain CLI parameters, for example `--help`, may be captured by
`tbot` even if intended to be passed to the wrapped `tsh`. A `--` argument can
be used to ensure all following arguments are passed to `tsh` and ignored by
`tbot`.

Additionally, be aware of the following limitations of `tbot db`:
 - `tbot db connect` requires a `tbot db login` for certain database types,
   like MySQL, so that additional connection parameters can be written to a
   local configuration file.
 - `tbot db env` is not fully supported.

### tbot proxy

Allows for access to Teleport resources via a local TLS proxy in single-port mode.
The `tbot proxy` command acts as a wrapper for `tsh proxy` to provide local proxy functionality for various protocols.

> Note that tsh must be installed to make use of this command.

Reviewer: ```suggestion
Note that `tsh` must be installed to make use of this command.
```

| `--destination-dir` | Path to the Machine ID destination dir that should be used for authentication. Required.                 |
| `--proxy`           | The `host:port` of the Teleport Proxy Service through which resources will be accessed. Required.        |
| `--cluster`         | The name of the cluster on which resources should be accessed. Extracted from the bot identity if unset. |

All other flags and arguments are passed directly to `tsh proxy ...`, along
with authentication parameters to use the Machine ID identity to skip `tsh`'s
login step.

Note that certain CLI parameters, for example `--help`, may be captured by
`tbot` even if intended to be passed to the wrapped `tsh`. A `--` argument can
be used to ensure all following arguments are passed to `tsh` and ignored by
`tbot`

#### Examples

**Example using OpenSSH**

The following command forwards standard input and output over a proxy suitable for use as an OpenSSH `ProxyCommand` for SSH access:

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 ssh alice@node:3022
```

  In this case:
   - `alice` is the remote username
   - `node` is the Teleport Node name

>    - `3022` is the remote SSH port, which is 3022 for Nodes running the Teleport

Reviewer: ```suggestion
   - `3022` is the remote SSH port, which is `3022` for Nodes running the Teleport
```

| `-a/--auth-server`   | Address of the Teleport Auth Server (on-prem installs) or Teleport Cloud tenant.               |
| `--token`            | A bot join token, if attempting to onboard a new bot; used on first connect. Can also be an absolute path to a file containing the token. |
| `--ca-pin`           | CA pin to validate the Teleport Auth Server; used on first connect.                            |
| `--data-dir`         | Directory to store internal bot data. In production environments access to this directory should be limited only to an isolated linux user as an owner with `0600` permissions. |
| `--destination-dir`  | Directory to write short-lived machine certificates.                                           |
| `--certificate-ttl`  | TTL of short-lived machine certificates.                                                       |
| `--renewal-interval` | Interval at which short-lived certificates are renewed; must be less than the certificate TTL. |
| `--join-method`      | Method to use to join the cluster. Can be `token` or `iam`.                                    |
| `--oneshot`          | If set, quit after the first renewal.

> teleport start \

Reviewer: I think we should add this to a `code` fence with a `$` prompt.

   --auth-server=example.teleport.sh:443
```

<Tabs>

> <TabItem scope={["cloud"]} label="Teleport Cloud">

Reviewer: Did you also add closing tags?
